apiserver/facades/client/application/application.go

```diff
@@ -1357,6 +1352,40 @@ func (api *APIBase) AddRelation(ctx context.Context, args params.AddRelation) (_
 			args.Endpoints[0], args.Endpoints[1], err,
 		)
 	}
+
+	if len(args.ViaCIDRs) > 0 {
+		// We only support relation networks for CMR, so we need to check that
+		// the newly created relation is indeed a CMR.
+		relationKey, err := corerelation.NewKey(
+			[]corerelation.EndpointIdentifier{
+				ep1.EndpointIdentifier(),
+				ep2.EndpointIdentifier(),
+			})
+		if err != nil {
+			return params.AddRelationResults{}, internalerrors.Errorf(
+				"creating relation key for relation between endpoints %q and %q: %w",
+				args.Endpoints[0], args.Endpoints[1], err,
+			)
+		}
+		isCMR, err := api.crossModelRelationService.IsRelationCrossModel(ctx, relationKey)
+		if err != nil {
+			return params.AddRelationResults{}, internalerrors.Errorf(
+				"checking if relation between endpoints %q and %q is cross-model: %w",
+				args.Endpoints[0], args.Endpoints[1], err,
+			)
+		}
+		if !isCMR {
+			return params.AddRelationResults{}, errors.NotSupportedf("integration via subnets for non cross model relations")
+		}
```

Comment:
Ideally, as much of this as possible should be pushed into the service layer

Reply:
I agree with @jack-w-shaw here, this should be pushed into the `AddRelationNetworkEgress`